internal/controllers/clusterextension_controller.go

```diff
@@ -489,53 +601,72 @@ func (r *ClusterExtensionReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	//   owned by the ClusterExtension.
-	existingBundleDeployment, err := r.existingBundleDeploymentUnstructured(ctx, desiredBundleDeployment.GetName())
-	if client.IgnoreNotFound(err) != nil {
-		return err
-	}
+func mapOwneeToOwnerHandler(cl client.Client, log logr.Logger, owner client.Object) crhandler.EventHandler {
+	return crhandler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+		ownerGVK, err := apiutil.GVKForObject(owner, cl.Scheme())
+		if err != nil {
+			log.Error(err, "map ownee to owner: lookup GVK for owner")
+			return nil
+		}
+		owneeGVK, err := apiutil.GVKForObject(obj, cl.Scheme())
+		if err != nil {
+			log.Error(err, "map ownee to owner: lookup GVK for ownee")
+			return nil
+		}
 
-	// If the existing BD already has everything that the desired BD has, no need to contact the API server.
-	// Make sure the status of the existingBD from the server is as expected.
-	if equality.Semantic.DeepDerivative(desiredBundleDeployment, existingBundleDeployment) {
-		*desiredBundleDeployment = *existingBundleDeployment
-		return nil
-	}
+		type ownerInfo struct {
+			key types.NamespacedName
+			gvk schema.GroupVersionKind
+		}
+		var oi *ownerInfo
 
-	return r.Client.Patch(ctx, desiredBundleDeployment, client.Apply, client.ForceOwnership, client.FieldOwner("operator-controller"))
-}
+		for _, ref := range obj.GetOwnerReferences() {
+			gv, err := schema.ParseGroupVersion(ref.APIVersion)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("map ownee to owner: parse ownee's owner reference group version %q", ref.APIVersion))
+				return nil
+			}
+			refGVK := gv.WithKind(ref.Kind)
+			if refGVK == ownerGVK && ref.Controller != nil && *ref.Controller {
+				oi = &ownerInfo{
+					key: types.NamespacedName{Name: ref.Name},
+					gvk: ownerGVK,
+				}
+				break
+			}
+		}
+		if oi == nil {
+			return nil
+		}
 
-func (r *ClusterExtensionReconciler) existingBundleDeploymentUnstructured(ctx context.Context, name string) (*unstructured.Unstructured, error) {
-	existingBundleDeployment := &rukpakv1alpha2.BundleDeployment{}
-	err := r.Client.Get(ctx, types.NamespacedName{Name: name}, existingBundleDeployment)
-	if err != nil {
-		return nil, err
-	}
-	existingBundleDeployment.APIVersion = rukpakv1alpha2.GroupVersion.String()
-	existingBundleDeployment.Kind = rukpakv1alpha2.BundleDeploymentKind
-	unstrExistingBundleDeploymentObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(existingBundleDeployment)
-	if err != nil {
-		return nil, err
-	}
-	return &unstructured.Unstructured{Object: unstrExistingBundleDeploymentObj}, nil
+		if err := cl.Get(ctx, oi.key, owner); client.IgnoreNotFound(err) != nil {
+			log.Info("map ownee to owner: get owner",
+				"ownee", client.ObjectKeyFromObject(obj),
+				"owneeKind", owneeGVK,
+				"owner", oi.key,
+				"ownerKind", oi.gvk,
+				"error", err.Error(),
+			)
+			return nil
+		}
```

Comment:
This was something done in rukpak, to log the owner of the `oi.key` object.
See https://github.com/operator-framework/rukpak/blob/58305404ab23a4117acb2726c515f44beb6a3b7f/pkg/util/util.go#L93-L102

Reply:
Reopening. In rukpak there was [this check](https://github.com/operator-framework/rukpak/blob/58305404ab23a4117acb2726c515f44beb6a3b7f/pkg/util/util.go#L103-L105) for the owner's provisioner class and we do not have it here. So this seems unnecessary.

Also logging only happens in case of an error on a get (anything other than 404):
* Most likely `owner` will be an empty struct anyway on error and logging fields will be empty strings as result
* If we do not query owner here - we do not have to log an issue here. I would expect 404 (or anything else) to be handled on reconcile anyway.

Reply:
I see the usefulness of this check and logging of the owner in case the obj that caused reconciliation is not existing on cluster. 

If owner is not found - we skip. If not, it cannot be empty, because in line 616 we first query the owner's GVK. Before adding a reconcile request, this is ensuring that the object is not being modified b/w the time we get the owner info (in line 616), to when we add the reconcile request to the queue. If there is any other error apart from 404, it means that the object has been updated/ partially modified and we return without queuing. By doing this we are ensuring that the owner is current before putting in a request for reconcile.

Reply:
I would keep this check as is given it's not just for logging.